sklearn/metrics/_regression.py

```diff
@@ -137,6 +140,63 @@ def _check_reg_targets(y_true, y_pred, multioutput, dtype="numeric", xp=None):
     return y_type, y_true, y_pred, multioutput
 
 
+def _check_reg_targets_with_floating_dtype(
+    y_true, y_pred, sample_weight, multioutput, xp=None
+):
+    """Ensure that y_true and y_pred correspond to the same regression task.
```

Comment:
Should we mention `sample_weight` as well here? The line is already quite long, but it feels like we should mention it, because this is a difference to `_check_reg_targets`. Not quite sure how though